### PR TITLE
MidiRawClient: Fix setting useless param

### PR DESCRIPTION
**Pull request by JohannesLorenz**
MidiChannelPressure and MidiProgramChange have only one param, so only
set this one param for those.

### CODE REVIEW

File: src/core/midi/MidiClient.cpp [hunk 1]
```diff
@@ -221,10 +221,12 @@ void MidiClientRaw::parseData( const unsigned char c )
 		case MidiNoteOff:
 		case MidiNoteOn:
 		case MidiKeyPressure:
+			// set velocity of key
+			m_midiParseData.m_midiEvent.setVelocity(m_midiParseData.m_buffer[1]);
```
Comment (PhysSong) [inline src/core/midi/MidiClient.cpp:225]: There really should be an explicit comment or annotation when falling through in a `switch`-`case`.

File: src/core/midi/MidiClient.cpp [hunk 2]
```diff
@@ -221,10 +221,12 @@ void MidiClientRaw::parseData( const unsigned char c )
...
 		case MidiChannelPressure:
 		case MidiProgramChange:
+			// set key (or for ChannelPressure/ProgramChange: set 1st param)
 			m_midiParseData.m_midiEvent.setKey(m_midiParseData.m_buffer[0]);
```
Comment (PhysSong) [inline src/core/midi/MidiClient.cpp:229]: I think `m_midiParseData.m_midiEvent.setParam(0, m_midiParseData.m_buffer[0]);` is semantically more correct, but I'll leave the decision up to you.

**Comment by PhysSong** [comment]
I think it would be better to eliminate the fall-through completely if we use `setParam` directly in two cases, which doesn't clearly represent the meaning for three cases above where it is for note key.

**Review by PhysSong — APPROVED** [review]

**Comment by JohannesLorenz** [comment]
For the record, the CI error also occurs in other PRs.

This PR awaits 2nd approval and testing.

**Comment by PhysSong** [comment]
@JohannesLorenz I think it's safe to merge without 2nd review.

**Pull request merged**